autoload/ale/list.vim

```diff
@@ -39,6 +39,13 @@ function! ale#list#SetLists(buffer, loclist) abort
     if (g:ale_open_list && !empty(a:loclist)) || g:ale_keep_list_window_open
         let l:winnr = winnr()
 
+        let l:reselect = 0
```

Comment:
Instead of the values `0`, `1`, and `2`, which don't really have any meaning, use a couple of Boolean variables which explain their purpose like `l:reset_visual_selection`.